frontend/components/GesturesToSound.js

```diff
@@ -0,0 +1,168 @@
+import React, {useCallback, useEffect, useState, useRef} from "react";
+import STYLES from "./summerPrototypes/ImageAnalysis.module.scss";
+import {getCookie} from "../common";
+
+
+const GesturesToSound = () => {
+    const canvasRef = useRef(null);
+    const [isGesturing, setIsGesturing] = useState(false);
+    const [currMouseCoords, setCurrMouseCoords] = useState([]);
+    const [allMouseCoords, setAllMouseCoords] = useState([]);
+    const [submitted, setSubmitted] = useState(false);
+    const [soundData, setSoundData] = useState(null);
+
+    const getCoords = (event) => {
+        if (!canvasRef.current) return;
+        const canvas = canvasRef.current;
+        const rect = canvas.getBoundingClientRect();
+        const scale = {
+            x: canvas.width / rect.width,
+            y: canvas.height / rect.height,
+        };
+        return {
+            x: (event.clientX - rect.left) * scale.x,
+            y: (event.clientY - rect.top) * scale.y,
+        };
+    };
+
+    const drawLine = (coords) => {
+        if (!canvasRef.current) return;
+        const canvas = canvasRef.current;
+        const context = canvas.getContext("2d");
+        if (context) {
+            context.beginPath();
+            context.lineWidth = 5;
+            context.lineJoin = "round";
+            context.lineCap = "round";
+            context.globalCompositeOperation = "source-over";
+            context.moveTo(coords[0].x, coords[0].y);
+            let i;
```

Comment:
Nitpick: you should probably write this `for (let i = 0 ...`, rather than declaring `i` outside of the loop. In this case, there's no practical difference, but there is an actual difference: 

```
let i;
for (i = 0; i < 2; i++) {}
i === 1  // This evaluates to true: i is in the outer scope, and has been modified by the for loop
```

```
for (let i = 0; i < 2; i++) {}
i === 1  // This will actually crash, as i is not defined in the outer scope
```
In general, we want the second case, as scoping your variables as tightly as possible makes your code easier to understand and less likely to have unwanted side effects.